//
//
// Copyright (c) 2015, University of Michigan.
// All rights reserved.
//
// This source code is licensed under the BSD-style license found in the
// LICENSE file in the root directory of this source tree. An additional grant
// of patent rights can be found in the PATENTS file in the same directory.
//
//

//
// TODO:
//
// @author: Johann Hauswald
// @contact: jahausw@umich.edu
//

package store_mat;

message Map {
    required uint32 size = 1;
    repeated Pair pair = 2;
    }

message Pair {
    required string str = 1;
    required mat_data mat = 2;
    }

message mat_data {
    required uint32 row = 1;
    required uint32 col = 2;
    required uint32 flags = 3;
    required uint32 step = 4;
    required uint32 size = 5;
    required uint32 type = 6;
    repeated uint32 data = 7[packed=true];
    }
